app/components/UI/Earn/Views/EarnLendingDepositConfirmationView/EarnLendingDepositConfirmationView.test.tsx

```diff
@@ -258,21 +368,108 @@ describe('EarnLendingDepositConfirmationView', () => {
-        from: '0xC4966c0D659D99699BFD7EB54D8fafEE40e4a756',
-        to: AAVE_V3_ETHEREUM_MAINNET_POOL_CONTRACT_ADDRESS,
-        value: '0',
-      },
-      {
+    expect(
+      Engine.context.EarnController.executeLendingDeposit,
+    ).toHaveBeenCalledWith({
+      amount: '5000000',
+      protocol: 'AAVE v3',
+      underlyingTokenAddress: MOCK_USDC_MAINNET_ASSET.address,
+      gasOptions: {},
+      txOptions: {
         deviceConfirmedOn: 'metamask_mobile',
         networkClientId: 'mainnet',
         origin: 'metamask',
         type: 'lendingDeposit',
       },
+    });
+  });
+
+  it('handles transaction error during approval flow', async () => {
+    const routeParamsWithApproveAction = {
+      ...defaultRouteParams,
+      params: {
+        ...defaultRouteParams.params,
+        action: EARN_LENDING_ACTIONS.ALLOWANCE_INCREASE,
+      },
+    };
+
+    (useRoute as jest.Mock).mockReturnValue(routeParamsWithApproveAction);
+
+    mockAddTransaction.mockRejectedValue(new Error('Transaction failed'));
+
+    const { getByTestId } = renderWithProvider(
+      <EarnLendingDepositConfirmationView />,
+      { state: mockInitialState },
+    );
+
+    const approveButton = getByTestId(
+      CONFIRMATION_FOOTER_BUTTON_TEST_IDS.CONFIRM_BUTTON,
+    );
+
+    await act(async () => {
+      fireEvent.press(approveButton);
+    });
+
+    expect(
+      Engine.context.EarnController.executeLendingTokenApprove,
+    ).toHaveBeenCalled();
+  });
+
+  it('handles transaction error during deposit flow', async () => {
+    mockAddTransaction.mockRejectedValue(new Error('Transaction failed'));
+
+    const { getByTestId } = renderWithProvider(
+      <EarnLendingDepositConfirmationView />,
+      { state: mockInitialState },
+    );
+
+    const depositButton = getByTestId(
+      CONFIRMATION_FOOTER_BUTTON_TEST_IDS.CONFIRM_BUTTON,
     );
+
+    await act(async () => {
+      fireEvent.press(depositButton);
+    });
+
+    expect(
+      Engine.context.EarnController.executeLendingDeposit,
+    ).toHaveBeenCalled();
+  });
+
+  it('displays correct token information in the UI', () => {
+    const { getByText } = renderWithProvider(
+      <EarnLendingDepositConfirmationView />,
+      { state: mockInitialState },
+    );
+
+    expect(getByText('45 USDC')).toBeDefined();
+    expect(getByText('5 aUSDC')).toBeDefined();
+    expect(getByText('4.5%')).toBeDefined();
+  });
+
+  it('displays correct amount information in the UI', () => {
+    const { getByText, getAllByText } = renderWithProvider(
+      <EarnLendingDepositConfirmationView />,
+      { state: mockInitialState },
+    );
+
+    expect(getAllByText('$4.99')).toBeDefined();
+    expect(getByText('5 aUSDC')).toBeDefined();
+    expect(getByText('5 USDC')).toBeDefined();
+  });
+
+  it('handles missing route params gracefully', () => {
```

Comment:
C: avoid "handle" and explain your definition of grace.
```suggestion
  it('handles missing route params gracefully', () => {
```